slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -56,11 +56,16 @@ ::grpc::Status SLAMServiceImpl::GetMap(ServerContext *context,
             return grpc::Status(grpc::StatusCode::UNAVAILABLE, oss.str());
         }
     } else if (mime_type == "pointcloud/pcd") {
-        LOG(ERROR) << "GetMap for mime_type \"pointcloud/pcd\" is not yet "
-                      "implemented.\n";
-        return grpc::Status(
-            grpc::StatusCode::UNIMPLEMENTED,
-            "GetMap for mime_type \"pointcloud/pcd\" is not yet implemented.");
+        std::stringbuf buffer;
+        bool has_points = ExtractPointCloudToBuffer(buffer);
+        if (has_points) {
+            common::v1::PointCloudObject *pco = response->mutable_point_cloud();
+            pco->set_point_cloud(buffer.str());
+        } else {
+            LOG(ERROR) << "map pointcloud does not have points yet\n";
```

Comment:
[nit] I think we don't need `\n` at the end of log lines.

Reply:
right, forgot about that!